admin/RemoveEmptyAccounts.pl

```diff
@@ -0,0 +1,238 @@
+                    FROM label_rating_raw
+                   WHERE label_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM place_rating_raw
+                   WHERE place_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM recording_rating_raw
+                   WHERE recording_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM release_group_rating_raw
+                   WHERE release_group_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM work_rating_raw
+                   WHERE work_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM annotation
+                   WHERE annotation.editor = e.id)
+  AND NOT EXISTS (   SELECT 1
+                       FROM editor_collection ec
+                      WHERE ec.editor = e.id
+                        AND ((
+                                  (type != 5 OR name != 'Attending')
+                                  AND
+                                  (type != 6 OR name != 'Maybe attending')
+                            )
+                            OR EXISTS (SELECT 1
+                                         FROM editor_collection_event ece
+                                        WHERE ece.collection = ec.id)))
+  AND NOT EXISTS (   SELECT 1
+                       FROM editor_collection ec
+                       JOIN editor_collection_collaborator ecc
+                         ON ecc.collection = ec.id
+                      WHERE ec.editor = e.id)
+  AND NOT EXISTS (   SELECT 1
+                       FROM editor_collection_collaborator ecc
+                      WHERE ecc.editor = e.id)
+             AND (   last_login_date IS NULL
+                  OR last_login_date < NOW() - INTERVAL '12 months')
+  SQL
+
+$c_cursor->sql->begin;
+$c_cursor->sql->do("DECLARE cursor NO SCROLL CURSOR FOR $query");
+
+my $pull = sub {
+    $c_cursor->sql->select("FETCH FORWARD $limit FROM cursor");
+};
+while ($pull->()) {
+    $c->sql->begin;
+    my @row;
+    while (@row = $c_cursor->sql->next_row) {
+        my ($editor_id, $editor_name) = @row;
+
+        if ($dry_run) {
+            print 'Removing account ' . $editor_name . " (dry run)\n";
+        } else {
+            print 'Removing account ' . $editor_name . "\n";
+            # Remove preferences
+            $c->sql->do(
+                'DELETE FROM editor_preference WHERE editor = ?',
+                $editor_id
+            );
+            # Remove languages
+            $c->sql->do(
+                'DELETE FROM editor_language WHERE editor = ?',
+                $editor_id
+            );
+            # Remove any subscriptions *to* this editor
+            $c->sql->do(<<~'SQL', $editor_id);
+                DELETE FROM editor_subscribe_editor
+                      WHERE subscribed_editor = ?
+                SQL
+            # Remove any collections by the editor (they should be all empty)
```

Comment:
I can't actually find where we ensure that all the collections are empty, other than for `editor_collection_event`.